js/components/MilkdropWindow/index.js

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import screenfull from "screenfull";
+import butterchurn from "butterchurn";
+import reactionDiffusion2 from "butterchurn-presets/presets/converted/Geiss - Reaction Diffusion 2";
+
+class MilkdropWindow extends React.Component {
+  constructor() {
+    super();
+  }
+  componentDidMount() {
+    const scale = window.devicePixelRatio;
+    const analyserNode = this.props.analyser;
+    this.visualizer = butterchurn.createVisualizer(
+      analyserNode.context,
+      this._canvasNode,
+      {
+        width: this._canvasNode.width * scale,
+        height: this._canvasNode.height * scale
+      }
+    );
+    this.visualizer.connectAudio(analyserNode);
+    this.visualizer.loadPreset(reactionDiffusion2, 0);
+    this._renderViz();
+
+    require.ensure(["butterchurn-presets"], require => {
+      const butterchurnPresets = require("butterchurn-presets");
+      const presets = butterchurnPresets.getPresets();
+      const presetKeys = Object.keys(presets);
+      this.cycleInterval = setInterval(() => {
+        const presetIdx = Math.floor(presetKeys.length * Math.random());
+        const preset = presets[presetKeys[presetIdx]];
+        this.visualizer.loadPreset(preset, 2.7);
+      }, 15000);
+    });
+  }
+  componentWillUnmount() {
+    this._pauseViz();
+    this._stopCycling();
+  }
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.width !== prevProps.width ||
+      this.props.height !== prevProps.height
+    ) {
+      this._setRendererSize(this.props.width, this.props.height);
+    }
+  }
+  _renderViz() {
+    this.animationFrameRequest = requestAnimationFrame(() => this._renderViz());
+    this.visualizer.render();
+  }
+  _pauseViz() {
+    if (this.animationFrameRequest) {
+      window.cancelAnimationFrame(this.animationFrameRequest);
+      this.animationFrameRequest = null;
+    }
+  }
+  _stopCycling() {
+    if (this.cycleInterval) {
+      clearInterval(this.cycleInterval);
+      this.cycleInterval = null;
+    }
+  }
+  _setRendererSize(width, height) {
+    const scale = window.devicePixelRatio;
+    this._canvasNode.width = width;
+    this._canvasNode.height = height;
+    this.visualizer.setRendererAndTextureSize(
+      width,
+      height,
+      width * scale,
+      height * scale
+    );
+  }
+  _handleRequestFullsceen() {
+    if (screenfull.enabled) {
+      if (!screenfull.isFullscreen) {
+        screenfull.request(this._canvasNode);
+        this._setRendererSize(window.innerWidth, window.innerHeight);
+      } else {
+        screenfull.exit();
+        this._setRendererSize(this.props.width, this.props.height);
+      }
+    }
+  }
+  render() {
+    return (
+      <canvas
+        className="draggable"
+        ref={node => (this._canvasNode = node)}
+        onDoubleClick={() => this._handleRequestFullsceen()}
```

Comment:
I think you can just add:

```
<canvas
    // ...
    width={this.props.width}
    height={this.props.height}
    // ...
/>
```

And avoid having to set properties on the node directly in `_setRendererSize`

Reply:
Oh, just kidding. You need to set them directly in case it's full screen don't you. Ignore me!